core/domain/wipeout_service.py

```diff
@@ -1253,3 +1255,88 @@ def _pseudonymize_models_transactional(voiceover_application_models):
             voiceover_application_models[
                 i:i + feconf.MAX_NUMBER_OF_OPS_IN_TRANSACTION]
         )
+
+
+def _pseudonymize_blog_posts_models(pending_deletion_request):
+    """Pseudonymizes the blog post models for the user with user_id.
+
+    Args:
+        pending_deletion_request: PendingDeletionRequest. The pending
+            deletion request object to be saved in the datastore.
+    """
+    user_id = pending_deletion_request.user_id
+
+    # We want to preserve the same pseudonymous user ID on all the models
+    # related to one blog post. So we collect all the users' blog
+    # post models and blog post summary models then
+    # we generate a pseudonymous user ID and replace the user ID
+    # with that pseudonymous user ID in all the models.
+    blog_post_model_class = blog_post_models.BlogPostModel
+    blog_post_models_list = blog_post_model_class.query(
+        datastore_services.any_of(
+            blog_post_model_class.author_id == user_id)
+        ).fetch()
+    blogpost_ids = set([model.id for model in blog_post_models_list])
+
+    blog_post_summary_model_class = blog_post_models.BlogPostSummaryModel
+    blog_post_summary_models = blog_post_summary_model_class.query(
+        datastore_services.any_of(
+            blog_post_summary_model_class.author_id == user_id)
+        ).fetch()
+    blogpost_ids |= set([model.id for model in blog_post_summary_models])
+
+    _save_pseudonymizable_entity_mappings_to_different_pseudonyms(
+        pending_deletion_request, models.NAMES.blog_post, blogpost_ids)
+    # TODO(sll):Add wipeout for BlogPostsRightsModel after adding
+    # service layer.
```

Comment:
Not sure what this TODO means, also please create an issue for this and then do `TODO(#12345): …`

Reply:
Also I'm not really sure why we need to wait for the service layer.

Reply:
When user deletes the account, user id should be removed from the list of editor ids in the BlogPostsRightsModel. The function to remove the user id from editor id will be added in the blog_services.py file. We will use the same function here.

Reply:
No, let's keep these two things separate. The reason is that wipeout is in nature quite a separate component that doesn't depend on other parts of the codebase (not 100 % ofc), this is done to ensure that the deletion works properly and that changes to the other parts of the codebase do not affect the wipeout procedure that much.

Reply:
Done. PTAL.